plugins/chat/app/models/chat/message.rb

```diff
@@ -220,10 +220,12 @@ def rebake!(invalidate_oneboxes: false, priority: nil)
       blockquote
       emphasis
       replacements
-      heading
     ]
 
     def self.cook(message, opts = {})
+      rules = MARKDOWN_IT_RULES
+      rules << "heading" if opts[:user_id] && opts[:user_id] < 0
```

Comment:
```suggestion
      rules << "heading" if opts[:user_id] && opts[:user_id].negative?
```

super minor suggestion, don't mind if you find the other way clearer